docs/specs/instrumentation.md

```diff
@@ -30,11 +30,11 @@ For metric naming, Vector broadly follows the
 [Prometheus metric naming standards]. Hence, a metric name:
 
 * MUST only contain ASCII alphanumeric, lowercase, and underscores
-* MUST follow the `<name>_<unit>_[total]` template
+* MUST follow the `<namespace>_<name>_<unit>_[total]` template
+  * `namespace` represents a broad category of metrics (e.g., `component`, `buffer`, `topology`)
   * `name` is one or more words that describes the measurement (e.g., `memory_rss`, `requests`)
   * `unit` MUST be a single [base unit] in plural form, if applicable (e.g., `seconds`, `bytes`)
   * Counters MUST end with `total` (e.g., `disk_written_bytes_total`, `http_requests_total`)
-* MUST NOT contain a namespace since the `internal_metrics` source sets a configurable namespace
```

Comment:
Technically the metric should still not have a structured namespace; that is: the `namespace` field of `Metric` should be empty (or, at the least, it'll be overwritten by `internal_metrics).